test/ossl_store_test.c

```diff
@@ -128,8 +130,16 @@ static int test_store_get_params(int idx)
         return 0;
     }
 
-    if (!TEST_true(BIO_snprintf(uri, sizeof(uri), "%s/%s-params.pem",
-                                datadir, type)))
+    urifmt = "%s/%s-params.pem";
+#ifdef __VMS
+    {
+        char datadir_end = datadir[strlen(datadir) - 1];
```

Comment:
is `datadir_end != NULL && *datadir_end != '0' && strstr(datadir_end, ":]>") != NULL` possible?  If so, is the default for urifmt above correct?

Reply:
Mind rephrasing that?  I don't understand how what you wrote relates to the actual code.

Point is that if datadir ends with `:`, `]` or `>`, no slash should be added.  This is "just" a test program, and on VMS, the datadir *is* given in VMS directory format, so it will essentially end with `]`.  I haven't cared to do a more thorough check, 'cause we don't need that for our current uses...  for something more thorough, we should really have an internal path building library, like I remarked a few days ago: https://github.com/openssl/openssl/pull/15823#discussion_r655936568